tools/subgraph-v3/src/PoolMapping.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  DrainTo as DrainToEvent,
+  OwnershipTransferred as OwnershipTransferredEvent,
+  SetMax as SetMaxEvent,
+  SetScale as SetScaleEvent,
+  Withdraw as WithdrawEvent
+} from "../generated/Pool/Pool"
+import {
+  DrainTo,
+  OwnershipTransferred,
+  SetMax,
+  SetScale,
+  Withdraw
+} from "../generated/schema"
+
+export function handleDrainTo(event: DrainToEvent): void {
+  let entity = new DrainTo(
+    event.transaction.hash.toHex() + "-" + event.logIndex.toString()
+  )
+  entity.tokens = event.params.tokens
+  entity.dest = event.params.dest
+  entity.save()
+}
+
+export function handleOwnershipTransferred(
+  event: OwnershipTransferredEvent
+): void {
+  let entity = new OwnershipTransferred(
+    event.transaction.hash.toHex() + "-" + event.logIndex.toString()
```

Comment:
Don't know if every event can be downcasted to a basic event type, if it is possible, it could be interesting to extract a formatting method 
let entity = new OwnershipTransferred(formatId(event))